src/Core/Managed/Shared/Channel/InMemoryChannel.cs

```diff
@@ -133,7 +143,14 @@ public void Flush()
         /// <param name="timeout">Timeout interval to abort sending.</param>
         public void Flush(TimeSpan timeout)
         {
-            this.transmitter.Flush(timeout);
+            if (!isDisposed)
+            {
+                this.transmitter.Flush(timeout);
+            }
+            else
+            {
+                throw new ObjectDisposedException("InMemoryChannel", "The telemetry channel has been disposed.");
```

Comment:
I don't think we're looking at this the same way.  Architecturally, as far as InMemoryChannel is concerned, when it is disposed it should no longer be used because all of its resources should have been released and the instance should be invalid now and free for garbage collection.  The fact that it's internal implementation uses an InMemoryTransmitter instance which has a Flush method has no bearing on whether or not the InMemoryChannel instance was disposed.

But let's walk through your real world scenario:
- The user somehow gets the channel disposed, either they did it on purpose or it happened "accidentally".
- They call Flush on the channel and we'll pretend it does not check for isDisposed and it in turn calls into the transmitter to flush but the transmitter has also been disposed.
- The transmitter should throw some kind of exception but currently it may not.  At this point you expect the user to use an object which the .NET framework should have been free to reuse the memory from.  The garbage collector may not have collected it yet if the user is holding a reference to it somehow but it is also not considered a valid object state for use anymore.

This did make me find another issue, when the transmitter is disposed it should flush but instead any remaining telemetry is just quietly lost.  The existing code for the InMemoryTransmitter didn't dispose cleanly but the .NET framework design does not allow/support a user continuing to use a disposed object.